runtime/common/src/migrations/local_currency.rs

```diff
@@ -0,0 +1,440 @@
+			let mut weight = T::DbWeight::get().reads(2);
+
+			if let Some(true) = check_local_coupling::<T>(from, to) {
+				pallet_pool_system::Pool::<T>::mutate(TargetPoolId::get(), |maybe_pool| {
+					if let Some(pool) = maybe_pool {
+						pool.currency = to.into();
+					}
+				});
+				weight.saturating_accrue(T::DbWeight::get().writes(1));
+				log::info!("{LOG_PREFIX} Migrated pool currency");
+			} else {
+				log::info!("{LOG_PREFIX} Skipping pool currency migration");
+			}
+
+			weight
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+			assert!(
+				check_local_coupling::<T>(CurrencyOut::get().into(), CurrencyIn::get().into())
+					.unwrap()
+			);
+
+			let pool =
+				pallet_pool_system::Pool::<T>::get(TargetPoolId::get()).expect("Pool should exist");
+			assert!(pool.currency == CurrencyOut::get().into());
+
+			log::info!("{LOG_PREFIX} PRE UPGRADE: Finished");
+
+			Ok(vec![])
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(_: Vec<u8>) -> Result<(), DispatchError> {
+			let pool =
+				pallet_pool_system::Pool::<T>::get(TargetPoolId::get()).expect("Pool should exist");
+			assert!(pool.currency == CurrencyIn::get().into());
+
+			log::info!("{LOG_PREFIX} POST UPGRADE: Finished");
+
+			Ok(())
+		}
+	}
+
+	fn check_local_coupling<T>(from: CurrencyId, to: CurrencyId) -> Option<bool>
+	where
+		T: pallet_pool_system::Config
+			+ orml_asset_registry::Config<CustomMetadata = CustomMetadata, AssetId = CurrencyId>,
+		<T as orml_asset_registry::Config>::AssetId: From<CurrencyId>,
+	{
+		let to_meta = <orml_asset_registry::Pallet<T> as Inspect>::metadata(&to);
+		let from_meta = <orml_asset_registry::Pallet<T> as Inspect>::metadata(&from);
+
+		match (from_meta, to_meta) {
+			(Some(meta), Some(_)) => {
+				let to_currency_id: CurrencyId = meta
+					.additional
+					.local_representation
+					.map(|x| x.try_into().ok())??;
+				if to_currency_id == to {
+					Some(true)
+				} else {
+					log::error!(
+						"{LOG_PREFIX} CurrencyOut does not have CurrencyIn set as local currency"
+					);
+					Some(false)
+				}
+			}
+			(Some(_), None) => {
+				log::error!("{LOG_PREFIX} CurrencyIn is not registered");
+				None
+			}
+			_ => {
+				log::error!("{LOG_PREFIX} CurrencyOut is not registered");
+				None
+			}
+		}
+	}
+}
+
+pub mod add_bidirectional_trading_pair {
+	use cfg_primitives::Balance;
+	#[cfg(feature = "try-runtime")]
+	use cfg_traits::TokenSwaps;
+	use frame_support::dispatch::RawOrigin;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+
+	const LOG_PREFIX: &str = "AddBidirectionalTradingPair";
+
+	pub struct Migration<T, CurrencyOne, CurrencyTwo, MinOrderAmount>(
+		sp_std::marker::PhantomData<(T, CurrencyOne, CurrencyTwo, MinOrderAmount)>,
+	);
+
+	impl<T, CurrencyOne, CurrencyTwo, MinOrderAmount> OnRuntimeUpgrade
+		for Migration<T, CurrencyOne, CurrencyTwo, MinOrderAmount>
+	where
+		T: pallet_order_book::Config<
```

Comment:
If IIRC, `order-book` can be reset. So maybe this isn't needed. If can not be reset, then we should make another big and painful migration with the previous order-book refactor

Reply:
In `centrifuge`, only one order exists (with `buy_amount` 0, so no balance is held). In `altair`,  `order-book` does not exist.

Reply:
I added this migration as convenience to immediately be able to trade USDC variants with our local one. Normally, this should be done through Governance but that can only happen after RU 1025 because the `CurrencyId::LocalAsset` variant does not exist yet. Thus, this migration.

Reply:
Ah, then it makes a lot of sense, thanks!